src/watchdog.rs

```diff
@@ -41,103 +37,89 @@ pub enum TimerInterval {
 
 pub struct Options(ClockSource, TimerInterval);
 
-pub trait State {}
-
-pub struct Enabled;
-
-pub struct Disabled;
-
-impl State for Enabled {}
+pub struct Enabled {
+    _marker: PhantomData<*const ()>,
+}
+pub struct Disabled {
+    _marker: PhantomData<*const ()>,
+}
 
-impl State for Disabled {}
+pub struct WatchdogTimer<S> {
+    wdt_a: WDT_A,
+    _state: PhantomData<S>,
+}
 
-struct WDT;
+impl WatchdogTimer<()> {
+    pub fn new(wdt_a: WDT_A) -> WatchdogTimer<Enabled> {
+        WatchdogTimer { wdt_a, _state: PhantomData }
+    }
+}
 
-impl WDT {
+impl<S> WatchdogTimer<S> {
     const WDT_COUNTER_CLEAR: u16 = 0x0008;
     const WDT_MODE_SELECT: u16 = 0x0010;
     const WDT_CONTROL_HOLD: u16 = 0x0080;
     const WDT_PASSWORD: u16 = 0x5A00;
     const WDT_PASSWORD_MASK: u16 = 0x00FF;
 
-    fn with_password(bits: u16) -> u16 {
-        WDT::WDT_PASSWORD + (bits & WDT::WDT_PASSWORD_MASK)
+    #[inline(always)]
```

Comment:
Hey @reitermarkus , what's the reasoning behind making these functions compile as `inline`?

Reply:
Some module revs. in the new branch: 
https://github.com/msp432-rust/msp432p401r-hal/tree/Dev-update

*Including the Watchdog module

Reply:
This reduces the binary size, in debug mode at least. Most of these are helper methods which are only used once.